Sources/web3swift/Transaction/EIP2930Envelope.swift

```diff
@@ -0,0 +1,412 @@
+//  Package: web3swift
+//
+//  Created by Alex Vlasov.
+//  Copyright © 2018 Alex Vlasov. All rights reserved.
+//
+//  Support for EIP-2930 transaction type by Mark Loit 2022
+
+import Foundation
+import BigInt
+
+public struct EIP2930Envelope: EIP2718Envelope {
+    public let type: TransactionType = .eip2930
+
+    // common parameters for any transaction
+    public var nonce: BigUInt = 0
+    public var chainID: BigUInt? {
```

Comment:
Not fully sure what you mean here. The reason why it is wrapped is that for legacy (and existing code) reasons, chainID is an optional in the protocol, but for the new types it is not optional.

Reply:
Apple carries it forward in their style, mainly from ObjC. In any case this may get resolved somewhat depending on where we land with one of the other discussions.

Reply:
Oh let me describe it in more details btw. We have public `chainID` as a actually wrapper around private `internalChainID` and my suggestion is to rename latter to `_chainID`.

Reply:
This will all change now with the discussed `TransactionOptions`/`TrasactionParameters` update. In that I think I'm going to remove all parameters except a few basic ones that I expect any future transaction to have. (type, to, nonce, r, s, v - basically the basics for transmission and integrity) everything else will move to the new `TransactionParameters` struct for access, and be wrapped for access accordingly in the deprecation shim. In doing that, I can do away with the need to wrap `chainID` entirely, for the new transaction types.